src/Backups/BackupImpl.cpp

```diff
@@ -381,6 +381,10 @@ void BackupImpl::writeBackupMetadata()
             if (info.encrypted_by_disk)
                 *out << "<encrypted_by_disk>true</encrypted_by_disk>";
         }
+        else if (!info.reference_target.empty())
+            *out << "<reference_target>" << xml << info.reference_target << "</reference_target>";
```

Comment:
We could do that without changing the format of backups - just convert every "reference" to a normal `BackupFileInfo` info around [this line](https://github.com/ClickHouse/ClickHouse/pull/56460/files#diff-b2ae7bff257f13a78053912db7fd183c8efdaf06478b020f2df0848c532748e7L77) and then allow the [deduplication by checksum](https://github.com/ClickHouse/ClickHouse/pull/56460/files#diff-b2ae7bff257f13a78053912db7fd183c8efdaf06478b020f2df0848c532748e7R120-R134) do its job.

Reply:
Can't we disable the deduplication by size and checksum?

Reply:
>Can't we disable the deduplication by size and checksum?

It's disabled only for plain backups.